src/ripple/protocol/impl/STTx.cpp

```diff
@@ -183,18 +183,19 @@ bool STTx::checkSign(bool allowMultiSign) const
             // at the SigningPubKey.  It it's empty we must be
             // multi-signing.  Otherwise we're single-signing.
             Blob const& signingPubKey = getFieldVL (sfSigningPubKey);
-            sigGood = signingPubKey.empty () ?
+            ret = signingPubKey.empty () ?
                 checkMultiSign () : checkSingleSign ();
         }
         else
         {
-            sigGood = checkSingleSign ();
+            ret = checkSingleSign ();
         }
     }
     catch (std::exception const&)
     {
+        ret = {false, "Internal signature check failure."};
```

Comment:
Thanks for asking about that.  My initial take is that I prefer to leave the initialization at the top of the function not because it's needed now, but as a stop gap when maintenance happens. And the initialization is cheap, since it just sets a bool value and builds an empty string.

Then I realized that a default constructed `pair<bool, string>` will get exactly the initialization I specified.  So the initialization at the top is inevitable (although I did spell out exactly what I want).

So I think I'm inclined to leave it as is, unless you object.